src/include/sof/audio/mfcc/mfcc_comp.h

```diff
@@ -125,8 +125,8 @@ struct mfcc_state {
 	int low_freq;
 	int high_freq;
 	int sample_rate;
-	int waiting_fill:1; /**< booleans */
-	int prev_samples_valid:1;
+	unsigned int waiting_fill:1; /**< booleans */
+	unsigned int prev_samples_valid:1;
```

Comment:
let's just make them `bool` to avoid bit-operations. This will waste 4 bytes (unless the compiler decides to make them different size) but will be prettier, atomic, and avoid issues like the present one